### PR TITLE
Fix Identity Azure CLI Cred process exectuion

### LINKED ISSUE #9058: I tried the 1.1.0-beta.2. I consistently get that the process doesn't exit.
I tried the 1.1.0-beta.2. I consistently get that the process doesn't exit.

```
Caused by: java.lang.RuntimeException: Tried EnvironmentCredential, ManagedIdentityCredential, SharedTokenCacheCredential, AzureCliCredential but failed to acquire a token for any of them. Please verify the environment for either of them and see more details in the causes below.
        at com.azure.identity.ChainedTokenCredential.lambda$getToken$3(ChainedTokenCredential.java:51)
        at reactor.core.publisher.MonoDefer.subscribe(MonoDefer.java:44)
        ... 65 more
Caused by: java.lang.IllegalThreadStateException: process hasn't exited
        at java.base/java.lang.ProcessImpl.exitValue(ProcessImpl.java:522)
        at com.azure.identity.implementation.IdentityClient.authenticateWithAzureCli(IdentityClient.java:209)
        at com.azure.identity.AzureCliCredential.getToken(AzureCliCredential.java:35)
        at com.azure.identity.ChainedTokenCredential.lambda$getToken$1(ChainedTokenCredential.java:40)
        at reactor.core.publisher.FluxFlatMap$FlatMapMain.onNext(FluxFlatMap.java:378)
        at reactor.core.publisher.FluxIterable$IterableSubscription.slowPath(FluxIterable.java:243)
        ... 57 more
```

This on OSX 10.15 19A583 and openjdk 11.0.5.

_Originally posted by @sitase in https://github.com/Azure/azure-sdk-for-java/issues/8269#issuecomment-598704197_

### PR DESCRIPTION
**Pull request by g2vinay**
Fixes #9058

### CODE REVIEW

File: sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClient.java [hunk 1]
```diff
@@ -206,6 +206,8 @@
             }
             String processOutput = output.toString();
 
+            process.waitFor();
```
Comment (jianghaolu) [inline sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClient.java:209]: Should we have a timeout?

Comment (jianghaolu) [reply sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClient.java:209]: We can probably make an educated guess of the longest possible time the sub process would take and fail after that?

Comment (schaabs) [reply sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClient.java:209]: agreed this must have some default timeout. In .NET I set this to 10 seconds.

File: sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClient.java [hunk 2]
```diff
@@ -227,6 +229,8 @@
             token = new IdentityToken(accessToken, expiresOn, options);
         } catch (IOException e) {
             throw logger.logExceptionAsError(new IllegalStateException(e));
+        } catch (InterruptedException e) {
+            throw logger.logExceptionAsError(new IllegalStateException(e));
```
Comment (srnagar) [inline sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClient.java:233]:
```suggestion
        } catch (IOException  | InterruptedException e) {
            throw logger.logExceptionAsError(new IllegalStateException(e));
        } 
```

**Comment by g2vinay** [comment]
> Please add a timeout to waiting on the process exit.

added 10 seconds timeout

File: sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClient.java (hunk 1 above)
Comment (g2vinay) [reply sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClient.java]: added 10 seconds timeout

**Pull request merged**